core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -710,11 +710,11 @@ public int numLeafKeysIssued() {
     }
 
     @Override
-    public long getEarliestKeyCreationTime() {
+    public Instant getEarliestKeyCreationTimeInstant() {
         if (seed != null)
-            return seed.getCreationTimeSeconds();
+            return Instant.ofEpochSecond(seed.getCreationTimeSeconds());
         else
-            return getWatchingKey().getCreationTimeSeconds();
+            return Instant.ofEpochSecond(getWatchingKey().getCreationTimeSeconds());
```

Comment:
I would recommend the ternary operator here. With or without an intermediate (effectively final) `seconds` variable.

Reply:
Ok, rewritten.